src/App.js

```diff
@@ -1,26 +1,110 @@
-import React from 'react';
-import logo from './logo.svg';
+import React, { Component } from 'react';
 import './App.css';
+import contacts from './contacts.json';
 
-function App() {
-  return (
-    <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
-    </div>
-  );
+//set the state to 5 contacts (create the contactlist and set it to state)
+class App extends Component { 
+  state = {
+    contactList: contacts.slice(0, 5),
+  };
+
+// --- add the random contacts-------
+  addRandom = () => {
+    const copyList = [...this.state.contactList];
+    const random = contacts[Math.floor(Math.random() * contacts.length)];
+    if (!copyList.includes(random)) {
```

Comment:
If you `splice` the contact from contacts, and add it to contacts upon deletion, you do not need to validate for duplication between te copyList and the randomly selected contact.